From fcb6976b4485fa22c7259542acfe97dfe8787fe1 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 21 Sep 2021 12:30:22 +0800
Subject: [PATCH 12/30] Revert "drm/panfrost: Fix missing
 clk_disable_unprepare() on error in panfrost_clk_init()"

This reverts commit 54912723f16ba0cdd7db57bb0d10f33bfff636f4.
---
 drivers/gpu/drm/panfrost/panfrost_device.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/panfrost/panfrost_device.c b/drivers/gpu/drm/panfrost/panfrost_device.c
index c256929e859b..bf7c34cfb84c 100644
--- a/drivers/gpu/drm/panfrost/panfrost_device.c
+++ b/drivers/gpu/drm/panfrost/panfrost_device.c
@@ -60,8 +60,7 @@ static int panfrost_clk_init(struct panfrost_device *pfdev)
 	if (IS_ERR(pfdev->bus_clock)) {
 		dev_err(pfdev->dev, "get bus_clock failed %ld\n",
 			PTR_ERR(pfdev->bus_clock));
-		err = PTR_ERR(pfdev->bus_clock);
-		goto disable_clock;
+		return PTR_ERR(pfdev->bus_clock);
 	}
 
 	if (pfdev->bus_clock) {
-- 
2.27.0

